core/src/main/java/io/questdb/FolderMapping.java

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb;
+
+import io.questdb.std.Files;
+import io.questdb.std.LowerCaseCharSequenceObjHashMap;
+import io.questdb.std.str.Path;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class FolderMapping {
+
+    private static final char SEPARATOR = ',';
+
+    private final LowerCaseCharSequenceObjHashMap<CharSequence> map = new LowerCaseCharSequenceObjHashMap<>(4);
+    private int lo, hi, limit;
+
+    public void forEach(LowerCaseCharSequenceObjHashMap.CharSequenceObjConsumer<CharSequence> action) {
+        map.forEach(action);
+    }
+
+    public FolderMapping of(@Nullable CharSequence text, @NotNull Path path) throws ServerConfigurationException {
```

Comment:
We might want to handle situations when the path has a trailing slash and when it doesn't and store the path in a normalized way internally. Would be great to cover this with a test.

Reply:
good point, I imagine users might add a trailing slash to signify it is a folder. If so we should normalise it to not have it, will add tests.